src/test/java/de/cronn/validation_files_diff/action/ValidationFileAutomaticRenamerTest.java

```diff
@@ -0,0 +1,158 @@
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static de.cronn.assertions.validationfile.TestData.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ValidationFileAutomaticRenamerTest extends HeavyPlatformTestCase {
+
+	public void testRenameClass_renamesAllValidationFiles() throws IOException {
+		Path projectDir = createDefaultJavaModuleStructure();
+
+		Path openedTestFile = projectDir.resolve("src/test/Test.java");
+		Files.createFile(openedTestFile);
+		Files.writeString(openedTestFile, "public class Test {}");
+
+		Path validationFileDirectory = getValidationFileDirectory(projectDir);
+		Path outputFileDirectory = getOutputFileDirectory(projectDir);
+		Path tempFileDirectory = getTempFileDirectory(projectDir);
+
+		Files.createDirectories(validationFileDirectory);
+		Files.createDirectories(outputFileDirectory);
+		Files.createDirectories(tempFileDirectory);
+
+		createValidationFileInDirectories(projectDir, "Test_findStuff.json");
+
+		refreshFileSystem(projectDir);
+
+		PsiClass testClass = JavaPsiFacadeEx.getInstanceEx(getProject()).findClass("Test");
+
+		executeRename(testClass, "Test2");
+
+		assertValidationFileExistsInDirectories(projectDir, "Test2_findStuff.json");
+		assertValidationFileNotExistsDirectories(projectDir, "Test_findStuff.json");
+	}
+
+	public void testRenameMethod_renamesAllValidationFiles() throws IOException {
+		Path projectDir = createDefaultJavaModuleStructure();
+
+		Path openedTestFile = projectDir.resolve("src/test/Test.java");
+		Files.createFile(openedTestFile);
+		Files.writeString(openedTestFile, "public class Test {\n" +
+				"\tpublic void testSomething() {" +
+				"}\n" +
+				"}");
+
+		Path validationFileDirectory = getValidationFileDirectory(projectDir);
+		Path outputFileDirectory = getOutputFileDirectory(projectDir);
+		Path tempFileDirectory = getTempFileDirectory(projectDir);
+
+		Files.createDirectories(validationFileDirectory);
+		Files.createDirectories(outputFileDirectory);
+		Files.createDirectories(tempFileDirectory);
+
+		String validationFileName = "Test_testSomething.json";
+		String newValidationFileName = "Test_testSomething2.json";
+
+		createValidationFileInDirectories(projectDir, validationFileName);
+
+		refreshFileSystem(projectDir);
+
+		PsiMethod testMethod = JavaPsiFacadeEx.getInstanceEx(getProject()).findClass("Test").findMethodsByName("testSomething", false)[0];
+
+		executeRename(testMethod, "testSomething2");
+
+		assertValidationFileExistsInDirectories(projectDir, newValidationFileName);
+		assertValidationFileNotExistsDirectories(projectDir, validationFileName);
+	}
+
+	@NotNull
+	private static Path getTempFileDirectory(Path projectDir) {
+		return projectDir.resolve(TEST_TEMPORARY_DATA_DIR);
+	}
+
+	@NotNull
+	private static Path getValidationFileDirectory(Path projectDir) {
+		return projectDir.resolve(TEST_VALIDATION_DATA_DIR);
```

Comment:
Great idea to use `de.cronn.assertions.validationfile.TestData#TEST_VALIDATION_DATA_DIR`!

Why not use this directly also in `ValidationDiffProjectOptionsProvider` in order to prefill the same values that are used by the validation comparison itself?

However, no need to do this here - can be done separately at a later time.